src/libraries/Microsoft.PowerFx.Interpreter/Functions/ResolvedObjectHelpers.cs

```diff
@@ -20,19 +21,24 @@ public static FormulaValue RecalcFormulaInfo(RecalcFormulaInfo fi)
             return fi._value;
         }
 
-        public static FormulaValue OptionSet(OptionSet optionSet, IRContext irContext)
+        // $$$ - this allows behavior we really don't want ... can we block in the IR?
+        // OptionSets can act like Records: 
+        //   TimeUnit.Hours
+        //   If(true, TimeUnit).Hours
+        //   If(true, TextPosition, Align).Left
```

Comment:
Once we block these cases, then we can eliminate this function altogether. 

Worst case, if we did need to have a way to OptionSet to RecordValue, we could have a derived RV that just wraps optionSet with virtual GetField() . Rather than fully hydrating the record with all fields.